scripts/quests/otherAreas/An_Affable_Adamantking.lua

```diff
@@ -0,0 +1,321 @@
+-----------------------------------
+-- An Affable Adamantking
+-----------------------------------
+-- Log ID: 4, Quest ID: 107
+-- Beadeaux Zone     : !pos -45 24 60 147
+-- Peshi Yohnts      : !pos -6 -6 -145 241
+-- Beastmen's Banner : !pos 10 25 -50 148
+-----------------------------------
+-- !addquest 4 107
+-- !addItem 885 -- Turtle Shell
+-- !addItem 1637 -- Bugard Leather
+-- !addItem 15201 -- Quadav Barbut
+-----------------------------------
+require('scripts/globals/npc_util')
+-----------------------------------
+local qulunDomeID  = zones[xi.zone.QULUN_DOME]
+local windurstWoodsID  = zones[xi.zone.WINDURST_WOODS]
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.AN_AFFABLE_ADAMANTKING)
+
+quest.reward =
+{
+    title = xi.title.BRONZE_QUADAV,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.A_GENEROUS_GENERAL) ~= QUEST_ACCEPTED and
+                player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.AN_UNDERSTANDING_OVERLORD) ~= QUEST_ACCEPTED and
+                player:getQuestStatus(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.A_MORAL_MANIFEST) ~= QUEST_ACCEPTED and
+                player:getVar('BstHeadGearQuest_Conquest') < NextConquestTally() and
+                not player:hasItem(xi.item.DAVHUS_BARBUT) and
+                player:getMainLvl() >= 60
+        end,
+
+        [xi.zone.QULUN_DOME] =
+        {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    local pos = player:getPos()
+                    if
+                        pos.x >= 0 and
+                        pos.y >= 20 and
+                        pos.z >= 55 and
+                        pos.x <= 6 and
+                        pos.y <= 25 and
+                        pos.z <= 65
+                    then
+                        return 60
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [60] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player) -- begin quest
+                    elseif option == 1 then
+                        player:setVar('BstHeadGearQuest_Conquest', NextConquestTally()) -- if player declines they must wait until next conquest tally
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WINDURST_WOODS] =
+        {
+            ['Peshi_Yohnts'] =
+            {
+                onTrigger = function(player, npc)
```

Comment:
Looks like everything here is returning a progressEvent.  Is this intentional? That means this quest would be blocking potentially for anything else (unless it also has the same priority level return)

Reply:
A bit over zealous with the progressEvents, the status/check-in dialogues could be `event(#):importantOnce` to not be blocking 👍🏻